docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,592 @@
+        - *another-container*
+
+    systemd:
+      matchPatterns:
+        - chronyd.service
+        - firewalld.service
+        - sshd*.service
+
+status:
+  conditions: # NOTE: The conditions will probably need a full definition/documentation/possible adjustments
+  - lastTransitionTime: "2024-04-30T15:06:20Z"
+    message: All is good
+    reason: AsExpected
+    status: "False"
+    type: Progressing
+  - lastTransitionTime: "2024-04-30T15:06:19Z"
+    message: All is good
+    reason: AsExpected
+    status: "True"
+    type: Available
+  containers:
+    - id: fa31890ae222449abb00ff90ed801674
+      name: my-container
+      status: running
+      image: quay.io/myorg/myimage:v1.0.0
+      engine: podman
+  systemdUnits:
+    - name: chronyd.service
+      loadState: running     #Q: those fields need clatification, not obvious what they mean
+      activeState: active
+    - name: sshd.service
+      loadState: running
+      activeState: active
+  systemInfo: # The systemInfo field will likely change in future RFEs, i.e. measurements will probably
+              # go away or change format (to provide specific quotes for specific PCRs)
+    architecture: arm64
+    bootID: 87f7e27e-bdc0-42b1-b909-6dc81fe43ea2
+    machineID: fa31890ae222449abb00ff90ed801674
+    measurements:
+      pcr01: "0000000000000000000000000000000000000000000000000000000000000000"
+      ...
+      pcr16: "0000000000000000000000000000000000000000000000000000000000000000"
+    operatingSystem: linux
+  updatedAt: "2024-04-30T15:06:19Z"
+```
+
+##### Device state regarding fleet managment
+This diagram illustrates what the following sections describe in text.
+```mermaid
+stateDiagram-v2
+    state "<b>Individual Device</b><br/><b>owner</b>: nil (API writable)<br/><b>spec:</b> user writable // unmanaged" as Individual_Device
+    state "<b>Fleet Device</b><br/><b>owner</b>: the-matching-fleet<br/><b>spec</b>: fleet-controller writable // managed" as Fleet_Device
+    state "<b>Fleet Device (paused)</b><br/><b>owner</b>: the-matching-fleet<br/><b>spec</b>: user writable // management paused" as Fleet_Paused
+    [*] --> Individual_Device
+    Individual_Device --> Fleet_Device : matching fleet labels && owner == nil
+    Fleet_Device --> Individual_Device : no matching fleet labels
+    Fleet_Device --> Fleet_Paused : flightctl.io/fleet-controller=paused
+    Fleet_Paused--> Fleet_Device: ! flightctl.io/fleet-controller=paused
+    Fleet_Paused--> Individual_Device: no matching fleet labels
+```
+
+##### `/device/{name}` with no owner fleet (Individual Device)
+
+Devices with no matching labels to a fleet are manually managed and start with no `metadata.owner`
+reference; the fleet controller will not attempt to reconcile or configure the device in any way.
+
+A `metadata.owner` field can be written through the API to associate the device to an external
+owner (like ACM or AAP, etc..), and prevent the fleet controller from managing the device in any way.
```

Comment:
Should we call out `metadata. owner` has two potential writers the fleet controller and the user? It seemed inferred here.

Reply:
No, I think it's actually good. The fleet is just a controller of devices, and it uses the devices API, it claims ownership, also external agents claim ownership.

Ownership at some point could be established by TLS identity.. and the fleet controller on this design could talk to the API directly if we want. 

I think it's actually more isolated in terms of API/concerns because of that... external and internal owners work under the same rules, the fleet is just a controller of devices, device API nows nothing of the management (as pods...)

Reply:
Although, you'd be right on one thing. We are making the RBAC learn that when the fleet controller paused is set ... we keep the owner but still we let the users edit. Should we drop the owner to be fully consistent?

In the end we would still find the matching labels. hmmm